File: packages/ai_config/domain/catalog.py
from typing import Any, Dict, List, Optional

from pydantic import BaseModel

from .provider import Provider
from .model import Model


class Catalog(BaseModel):
    providers: List[Provider]
    models: List[Model]
    mode: Optional[str] = None
    defaults: Optional[Dict[str, Any]] = None
    meta: Optional[Dict[str, Any]] = None

File: packages/ai-config/src/domain/catalog.ts
import type { Provider } from './provider.js';
import type { Model } from './model.js';

export type CatalogMode = 'merge' | 'replace';

export interface Catalog {
  providers: Provider[];
  models: Model[];
  mode?: CatalogMode;
  defaults?: {
    provider?: string;
    model?: string;
    byContext?: Record<string, { provider?: string; model?: string; fallback?: string }>;
  };
  meta?: Record<string, unknown>;
}
